src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPoolEventSource.cs

```diff
@@ -3,117 +3,286 @@
+            public const string WorkerThreadAdjustmentSample = "Throughput={0};\nClrInstanceID={1}";
+            public const string WorkerThreadAdjustmentAdjustment = "AverageThroughput={0};\nNewWorkerThreadCount={1};\nReason={2};\nClrInstanceID={3}";
+            public const string WorkerThreadAdjustmentStats = "Duration={0};\nThroughput={1};\nThreadWave={2};\nThroughputWave={3};\nThroughputErrorEstimate={4};\nAverageThroughputErrorEstimate={5};\nThroughputRatio={6};\nConfidence={7};\nNewControlSetting={8};\nNewThreadWaveMagnitude={9};\nClrInstanceID={10}";
+            public const string IOEnqueue = "NativeOverlapped={0};\nOverlapped={1};\nMultiDequeues={2};\nClrInstanceID={3}";
+            public const string IO = "NativeOverlapped={0};\nOverlapped={1};\nClrInstanceID={2}";
+            public const string WorkingThreadCount = "Count={0};\nClrInstanceID={1}";
+        }
 
         // The task definitions for the ETW manifest
-        public static class Tasks
+        public static class Tasks // this name and visibility is important for EventSource
         {
-            public const EventTask WorkerThreadTask = (EventTask)16;
-            public const EventTask WorkerThreadAdjustmentTask = (EventTask)18;
+            public const EventTask ThreadPoolWorkerThread = (EventTask)16;
+            public const EventTask ThreadPoolWorkerThreadAdjustment = (EventTask)18;
+            public const EventTask ThreadPool = (EventTask)23;
+            public const EventTask ThreadPoolWorkingThreadCount = (EventTask)22;
         }
 
-        public static class Opcodes
+        public static class Opcodes // this name and visibility is important for EventSource
         {
-            public const EventOpcode WaitOpcode = (EventOpcode)90;
-            public const EventOpcode SampleOpcode = (EventOpcode)100;
-            public const EventOpcode AdjustmentOpcode = (EventOpcode)101;
-            public const EventOpcode StatsOpcode = (EventOpcode)102;
+            public const EventOpcode IOEnqueue = (EventOpcode)13;
+            public const EventOpcode IODequeue = (EventOpcode)14;
+            public const EventOpcode Wait = (EventOpcode)90;
+            public const EventOpcode Sample = (EventOpcode)100;
+            public const EventOpcode Adjustment = (EventOpcode)101;
+            public const EventOpcode Stats = (EventOpcode)102;
         }
 
-        public static class Keywords
+        public static class Keywords // this name and visibility is important for EventSource
         {
             public const EventKeywords ThreadingKeyword = (EventKeywords)0x10000;
+            public const EventKeywords ThreadTransferKeyword = (EventKeywords)0x80000000;
         }
 
         private PortableThreadPoolEventSource()
+            : base(
+                  new Guid(0xe13c0d23, 0xccbc, 0x4e12, 0x93, 0x1b, 0xd9, 0xcc, 0x2e, 0xee, 0x27, 0xe4),
```

Comment:
You should remove these parameters to the base class constructor and just let EventSource pick these up from the attribute.

Reply:
Other PRs have explicitly switched to using this base ctor to improve startup time / reduce reflection usage on startup, e.g.
https://github.com/dotnet/coreclr/pull/21714

Reply:
Not using the explicit Guid is harsh for startup since so much code needs to be Jitted

Reply:
Ah, yes that seems fine.  Just make sure that the GUID matches what EventSource would have generated for the name, so that tools still function properly if you don't specify the GUID, but instead specify the name.

Reply:
Wonder if there is a build verification step that can be added that checks the GUIDs against ones that would be generated?

Reply:
Definitely possible, but I'm not too concerned about this, just because it's essentially limited to EventSources in this repo.  If we wanted to find a way to fix this up more publicly, then I think we'd need to try and fix the API, or if we couldn't find a good way, then yes, a tool to validate.